modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/50_script_doc_values.yml

```diff
@@ -494,6 +750,58 @@ setup:
                             source: "doc['float'].value"
     - match: { hits.hits.0.fields.field.0: 3.1415927410125732 } # this ends up as a double
 
+    - do:
+        search:
+          rest_total_hits_as_int: true
+          body:
+            sort: [ { rank: asc } ]
+            script_fields:
+              field:
+                script:
+                  source: "Float.toString(field('float').get(-1))"
```

Comment:
You should be able to just do `toString` directly on `field('float').get(-1)` like `field('float').get(-1).toString()`